storm/src/main/storm/mesos/MesosNimbus.java

```diff
@@ -187,11 +197,16 @@ public void prepare(Map conf, String localDir) {
   }
 
   @SuppressWarnings("unchecked")
-  protected void initialize(Map conf, String localDir) throws Exception {
-    _conf = new HashMap();
-    _conf.putAll(conf);
+  void initializeMesosStormConf(Map conf, String localDir) {
+    mesosStormConf = new HashMap();
+    mesosStormConf.putAll(conf);
 
-    _state = new LocalStateShim(localDir);
+    try {
+      _state = new LocalStateShim(localDir);
+    } catch (IOException exp) {
+      // TODO(ksoundararaj) : Should we exit here?
```

Comment:
I think we should exit here, because without the LocalState (`_state`) this class cannot do its job.